plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -304,7 +301,7 @@ function OPDSBrowser:fetchFeed(item_url, headers_only)
             ["404"] = _("Catalog not found."),
             ["406"] = _("Cannot get catalog. Server refuses to serve uncompressed content."),
         }
-        text = error_message[tostring(code)] or T(_("Cannot get catalog. Server response status: %1."), status or code)
+        text = error_message[tostring(code or 0)] or T(_("Cannot get catalog. Server response status: %1."), status or code)
```

Comment:
This can't be right. You'd never see the fallback.

Reply:
It'd probably be better to split it up for clarity but something along these lines would come closer to the desired behavior.
```suggestion
        text = code and error_message[tostring(code)] or T(_("Cannot get catalog. Server response status: %1."), status or code)
```

Reply:
Oh wait, error_message[0] would still evaluate to nil? Meh, I find that unclear, so regardless of what it does I'd write it more clearly.

Reply:
The logic is:
(1) headers absence is not a special case
(2) codes 301 and 302 requires headers, otherwise fallback to the default message
(3) codes 401, 403, 404, 406 show specific messages
(4) other codes fallback to the default message
(5) no code case fallbacks to the default message

Reply:
> (4) other codes fallback to the default message
(5) no code case fallbacks to the default message

Right, but having a fake code 0 that does that isn't immediately obvious and requires reasoning. If the result of what I wrote isn't 100% identical that's not really the point of what I wrote but rather that it should be clearer so it can simply be read.

Reply:
Fake code 0 is for case "no code in response", to avoid tostring(nil) crash. I'll add a comment.

Reply:
Please take another look at my suggestion which requires neither reasoning nor a comment. ;-)

Reply:
Okay, thanks.